src/api/app/policies/status/report_policy.rb

```diff
@@ -0,0 +1,53 @@
+class Status::ReportPolicy < ApplicationPolicy
+  def initialize(user, record)
+    raise Pundit::NotAuthorizedError, 'record does not exist' unless record
+    @user = user
+    @record = record
+  end
+
+  def create?
+    return false if @user.blank?
+    return true if @user.is_admin?
+
+    checkable_containers = case @record.checkable
+                           when BsRequest
+                             @record.checkable.bs_request_actions.map { |action| request_target_object(action) }
+                           when Repository
+                             [@record.checkable.project]
+                           end
+
+    checkable_containers.all? { |container| write_permission_for_checkable?(container) }
+  end
+
+  def update?
+    create?
+  end
+
+  def destroy?
+    create?
+  end
+
+  def index?
+    true
+  end
+
+  def show?
+    index?
+  end
+
+  private
+
+  def write_permission_for_checkable?(container)
+    container &&
+      (container.relationships.users.pluck(:user_id).include?(@user.id) ||
+      GroupsUser.where(group: container.relationships.groups.pluck(:group_id), user: @user).exists?)
+  end
+
+  def request_target_object(action)
+    if action.target_package
```

Comment:
This code is repeated in https://github.com/openSUSE/open-build-service/pull/5862/commits/2d89be4ccaed5d27948c22e6c84cde4727b6d51d#diff-66e753326c6844bc8ae62de2dfe0cb35R44.... :sob: